### PR TITLE
Remove required

### PR DESCRIPTION
**Pull request by MongoCaleb**
### Jira

https://jira.mongodb.org/browse/DOCSP-30309

### Staged Changes

https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/remove-required/
but really nothing changed.

### CODE REVIEW

File: examples/dotnet/Examples/Examples.csproj [hunk 1]
```diff
@@ -4,6 +4,11 @@
 		<TargetFramework>net6.0</TargetFramework>
 		<LangVersion>default</LangVersion>
 		<IsPackable>false</IsPackable>
+		<Nullable>enable</Nullable>
+	</PropertyGroup>
+	<PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Debug|AnyCPU' ">
+	  <WarningLevel>4</WarningLevel>
+	  <NoWarn>1701;1702;8618;8602</NoWarn>
```
Comment (nirinchev) [inline examples/dotnet/Examples/Examples.csproj:11]: This suppression only applies for debug builds - you probably want it for release as well?

File: examples/dotnet/Examples/Schemas.cs [hunk 2]
```diff
@@ -57,13 +57,8 @@ public void TestSchemas()
...
             Assert.AreEqual(1, mixedConfig.Schema.Count);
-            ObjectSchema foo;
-            mixedConfig.Schema.TryFindObjectSchema("ClassA", out foo);
-            if (foo != null)
-            {
-                Property newProp;
-                Assert.IsTrue(foo.TryFindProperty("ThisIsNotInTheCSharpClass", out newProp));
-            }
+
+
```
Comment (nirinchev) [inline examples/dotnet/Examples/Schemas.cs:61]: Some leftover empty lines

File: examples/dotnet/Examples/Task.cs [hunk 3]
```diff
@@ -14,12 +16,10 @@ public class Item : RealmObject
         public User Assignee { get; set; }
 
         [MapTo("name")]
-        [Required]
-        public string Name { get; set; }
+        public string? Name { get; set; }
```
Comment (nirinchev) [inline examples/dotnet/Examples/Task.cs:19]: Should these be nullable? I guess either way is fine, it's just that I generally prefer to show non-nullable versions as examples as the nullability is extra information that is irrelevant for the example.

File: examples/dotnet/Examples/User.cs [hunk 4]
```diff
@@ -1,23 +1,16 @@
 ﻿using Realms;
 
+#nullable enable
```
Comment (nirinchev) [inline examples/dotnet/Examples/User.cs:3]: Do we need these pragmas?

File: examples/dotnet/Examples/WorkWithRealm.cs [hunk 5]
```diff
@@ -327,7 +327,7 @@ public NotificationUnsub()
             {
                 realm = Realm.GetInstance("");
             }
-            private IQueryable<Item> items = null;
+            private IQueryable<Item> items = new List<Item>().AsQueryable();
```
Comment (nirinchev) [inline examples/dotnet/Examples/WorkWithRealm.cs:330]: Writing `= null!` is a little shorter for cases like this one.

File: examples/dotnet/Examples/WriteExamples.cs [hunk 6]
```diff
@@ -207,7 +207,6 @@ public class WriteDog : RealmObject
         [MapTo("_id")]
         public int Id { get; set; }
 
-        [Required]
```
Comment (nirinchev) [inline examples/dotnet/Examples/WriteExamples.cs:210]: Some of these warnings are meaningful and some are not. For example, `WritePerson Owner` should be nullable - I'm kind of surprised Realm didn't complain about this one. The rest can either be ignored or you can declare a constructor to set them.

**Review by nirinchev — APPROVED** [review]

File: source/sdk/dotnet/model-data/data-types/sets.txt [hunk 7]
```diff
@@ -31,9 +31,11 @@ in which it existed. Therefore, a set of objects will never contain null objects
 However, sets of primitive types can contain null values. If you do not 
 want to allow null values in a set, then either use non-nullable types in 
 the set declaration (for example, use ``ISet<double>`` instead of 
-``ISet<double?>``), or add the 
+``ISet<double?>``). If you are using the older schema 
+type definition (your classes derive from the ``RealmObject`` base class),
+or you do not have nullability enabled, you will need to use the
 :ref:`[Required]<dotnet-required-optional-property>` attribute if the set  
-contains nullable reference types, such as ``string`` or ``byte[]``.
+contains nullable reference types, such as ``string`` or ``byte[]``. or add the
```
Comment (lindseymoore) [inline source/sdk/dotnet/model-data/data-types/sets.txt:38]:
```suggestion
contains nullable reference types, such as ``string`` or ``byte[]``.
```

Not sure if this thought was supposed to be completed or just needs to be deleted.

File: source/sdk/dotnet/model-data/data-types/dictionaries.txt [hunk 8]
```diff
@@ -30,7 +30,9 @@ A dictionary of objects can contain null objects. Likewise, dictionaries
 of primitive types can also contain null values. If you do not 
 want to allow null values in a dictionary, then either use non-nullable types in 
 the dictionary declaration (for example, use ``IDictionary<string, double>`` 
-instead of ``IDictionary<string, double?>``), or add the 
+instead of ``IDictionary<string, double?>``). If you are using the older schema 
+type definition (your classes derive from the ``RealmObject`` base class),
+or you do not have nullability enabled, you will need to use the
```
Comment (lindseymoore) [inline source/sdk/dotnet/model-data/data-types/dictionaries.txt:35]:
```suggestion
or you do not have nullability enabled, use the
```
Style nit: Present tense

File: source/sdk/dotnet/model-data/data-types/lists.txt [hunk 9]
```diff
@@ -41,7 +41,9 @@ where it existed. Therefore, a list of objects will never contain deleted object
 However, lists of primitive types can contain null values. If you do not 
 want to allow null values in a list, then either use non-nullable types in 
 the list declaration (for example, use ``IList<double>`` instead of 
-``IList<double?>``), or add the 
+``IList<double?>``). If you are using the older schema 
+type definition (your classes derive from the ``RealmObject`` base class),
+or you do not have nullability enabled, you will need to use the
```
Comment (lindseymoore) [inline source/sdk/dotnet/model-data/data-types/lists.txt:46]:
```suggestion
or you do not have nullability enabled, use the
```

**Review by lindseymoore — APPROVED** [review]
Purely a copy review. Once changes are made, LGTM!

**Pull request merged**